### PR TITLE
Refactor init process of WebFilter

### PR DESCRIPTION
**Pull request by emre-aydin**

### CODE REVIEW

File: src/test/webapp/WEB-INF/spring/hazelcast-spring.xml [hunk 1]
```diff
@@ -42,10 +42,6 @@
             <param-name>shutdown-on-destroy</param-name>
             <param-value>true</param-value>
         </init-param>
-        <init-param>
-            <param-name>client-config-location</param-name>
```
Comment (mesutcelik) [inline src/test/webapp/WEB-INF/spring/hazelcast-spring.xml]: why are those deleted in all configurations? `client-config-location` and `config-location`

File: src/main/java/com/hazelcast/web/WebFilterConfig.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        boolean useRequestParameter = getBoolean(filterConfig, properties, USE_REQUEST_PARAMETER, false);
+        boolean debug = getBoolean(filterConfig, properties, DEBUG, false);
+        Set<String> transientAttributes = getStringSet(filterConfig, properties, TRANSIENT_ATTRIBUTES);
+        String cookieName = getString(filterConfig, properties, COOKIE_NAME, "hazelcast.sessionId");
+        String cookieDomain = getString(filterConfig, properties, COOKIE_DOMAIN, null);
+        boolean cookieSecure = getBoolean(filterConfig, properties, COOKIE_SECURE, false);
+        boolean cookieHttpOnly = getBoolean(filterConfig, properties, COOKIE_HTTP_ONLY, false);
+        String cookiePath = getString(filterConfig, properties, COOKIE_PATH, null);
+
+        WebFilterConfig wfc = new WebFilterConfig();
```
Comment (mesutcelik) [inline src/main/java/com/hazelcast/web/WebFilterConfig.java]: why are we doing this? We can completely make `WebFilterConfig` statically initialized and used config util. by making all getters/fields static?

File: src/test/webapp/WEB-INF/spring/hazelcast-spring.xml (hunk 1 above)
Comment (emre-aydin) [inline src/test/webapp/WEB-INF/spring/hazelcast-spring.xml]: If P2P mode is used, `client-config-location` should not be present and vice versa for client mode. We now throw errors if there's an extra unused property (`client-config-location`) in this case.

File: src/main/java/com/hazelcast/web/WebFilterConfig.java (hunk 2 above)
Comment (emre-aydin) [inline src/main/java/com/hazelcast/web/WebFilterConfig.java]: There's a single place where we create `WebFilterConfig` on `WebFilter#init` method. As we advise users to define only a single instance of `WebFilter` in their `web.xml` file, there will effectively be a single `WebFilterConfig` instance and it will be initialised only once. 

Also, we let users configure `WebFilter` via Spring Framework and it wouldn't place nicely with `static` either. 

Another reason is it would make unit testing harder for classes that are using `WebFilterConfig`.

**Comment by mesutcelik** [comment]
👍 assuming moving config names are unchanged...

**Comment by emrahkocaman** [comment]
Nice clean-up!
👍
